.github/workflows/actions/early-exit-check/action.yml

```diff
@@ -9,7 +9,16 @@ runs:
     - name: Check location of changed files
       shell: bash
       run: |
-        change_count=$(git diff --name-only origin/main..HEAD | grep -E '^(app/.*)|(.yarn/.*)|(.github/workflows/.*)' | wc -l)
+        # On main branch, compare with the previous
+        # commit otherwise (for PRs) compare with the
+        # current commit.
+        if [ "${GITHUB_REF_NAME}" = "main" ]; then
+          end_ref="HEAD^"
+        else
+          end_ref="HEAD"
+        fi
+
+        change_count=$(git diff --name-only origin/main..${end_ref} | grep -E '^(app/.*)|(.yarn/.*)|(.github/workflows/.*)' | wc -l)
```

Comment:
can you make `origin/main` dynamic? We can use `${GITHUB_BASE_REF:-HEAD}` since `GITHUB_BASE_REF` is only provided for PRs, the `:-HEAD` will head the default to `HEAD` when not from a PR.

Reply:
Fixed, I had to include `origin` otherwise git complains if it being ambitious:

```console
Run # On main branch, compare with the previous
Comparing main with HEAD
fatal: ambiguous argument 'main..HEAD': unknown revision or path not in the working tree.
```